packages/devtools_app_shared/lib/src/utils/list_queue.dart

```diff
@@ -0,0 +1,103 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:collection';
+
+import 'package:flutter/foundation.dart';
+
+/// A [ChangeNotifier] that holds a [ListQueue] of data.
+///
+/// This class exposes methods to modify the underlying [ListQueue]. When the
+/// [ListQueue] is modified, listeners are notified.
+///
+/// When using this class, you should NOT modify the underlying [ListQueue]
+/// manually by calling [ListQueue] methods on this notifier's [value]. Doing so
+/// will result in listeners not being notified for changes to the [ListQueue].
+///
+/// This class is not a full implementation of the [ListQueue] class, but it
+/// exposes the majority of [ListQueue] methods that are useful for how this
+/// class is used in DevTools. New methods can be added to this class as needed.
+final class ListQueueValueNotifier<T> extends ChangeNotifier
```

Comment:
this should have tests. To not block this PR, I recommend putting this class under devtools_app/ and then adding a matching test file devtools_app/test/shared/ with a TODO and a tracking issue. Once this class is well tested we can consider adding to the public package devtools_app_shared/ for shared use.

Reply:
Done in 98d75b839